drivers/pwm/pwm_mspm0.c

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (c) 2024 Bang & Olufsen A/S, Denmark
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <ti/driverlib/dl_timera.h>
+#include <ti/driverlib/dl_timerg.h>
+#include <zephyr/drivers/pwm.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(pwm_mspm0, CONFIG_LOG_DEFAULT_LEVEL);
+
+#define DT_DRV_COMPAT ti_mspm0_pwm
+
+struct pwm_mspm0_config {
+	GPTIMER_Regs *timer;
+	const struct pinctrl_dev_config *pinctrl;
+	const uint32_t frequency;
+	const DL_Timer_ClockConfig clock_cfg;
+	const DL_TIMER_PWM_MODE pwm_mode;
+	const uint8_t cc_index;
+	const uint8_t direction;
+	const bool is_tima;
+	const bool four_cc;
+};
+
+static int pwm_mspm0_get_cycles_per_sec(const struct device *dev, uint32_t channel,
+					uint64_t *cycles)
+{
+	const struct pwm_mspm0_config *config = dev->config;
+	*cycles = config->frequency /
+		  ((config->clock_cfg.divideRatio + 1) * (config->clock_cfg.prescale + 1));
+	return 0;
+}
+
+static int pwm_mspm0_set_cycles(const struct device *dev, uint32_t channel, uint32_t period_cycles,
+				uint32_t pulse_cycles, pwm_flags_t flags)
+{
+	const struct pwm_mspm0_config *cfg = dev->config;
+	if (pulse_cycles > 100) {
+		LOG_DBG("Invalid duty cycles, range should be [0, 100]");
+		return -1;
+	}
+
+	// driver specific to stop the counter
+	if (flags == 1) {
+		DL_Timer_stopCounter(cfg->timer);
+		return 0;
+	}
```

Comment:
Is it not easier to react to a period of `0` meaning stop? Otherwise, I guess adding a `#define PWM_MSPM0_STOP (1 << 8)` to use is probably fine.

Actually, looking at the description of the `pwm_set_cycles` it states:
```
 * Passing 0 as @p pulse will cause the pin to be driven to a constant
 * inactive level.
 * Passing a non-zero @p pulse equal to @p period will cause the pin
 * to be driven to a constant active level.
```
So why not just follow that principle?